esrally/mechanic/launcher.py

```diff
@@ -175,6 +175,9 @@ def _prepare_env(self, car_env, node_name, java_home, t):
             self._set_env(env, "PATH", os.path.join(java_home, "bin"), separator=os.pathsep, prepend=True)
             # Don't merge here!
             env["JAVA_HOME"] = java_home
+        elif env.get("JAVA_HOME"):
+            # Assume that since java_home was not passed in, it should be unset in the environment if it exists
+            del env["JAVA_HOME"]
```

Comment:
Could we adjust the test in launcher_test#test_bundled_jdk_not_in_path() to fail with our current code, i.e. actually check that the JAVA_HOME env var remains unset even if set in os.environ?